src/components/battles/animations/shared/items/Item.ts

```diff
@@ -0,0 +1,86 @@
+import { IMoveAction, IMoveEffect, ITeamAndAction, Team } from "battlemovr";
+import { GeneralComponent } from "gamestartr";
+
+import { FullScreenPokemon } from "../../../../../FullScreenPokemon";
+import { IBattleInfo, IPokemon } from "../../../../Battles";
+import { IMenu } from "../../../../Menus";
+import { IThing } from "../../../../Things";
+
+/**
+ * Uses an item.
+ */
+export class Item<TGameStartr extends FullScreenPokemon> extends GeneralComponent<TGameStartr> {
+    /**
+     * Team and move being performed.
+     */
+    protected readonly teamAndAction: ITeamAndAction<IMoveAction>;
+
+    /**
+     * Item being used.
+     */
+    protected readonly item: string;
+
+    /**
+     * Pokemon being targeted.
+     */
+    protected readonly attacker: IPokemon;
+
+    /**
+     * Visually attaching Thing.
+     */
+    protected readonly attackerThing: IThing;
+
+    /**
+     * Pokemon using the item.
+     */
+    protected readonly defender: IPokemon;
+
+    /**
+     * Visually defending Thing being attacked.
+     */
+    protected readonly defenderThing: IThing;
+
+    /**
+     * Movement direction from the attacker to the defender.
+     */
+    protected direction: -1 | 1;
+
+    /**
+     * Battle display menu.
+     */
+    protected readonly menu: IMenu;
+
+    /**
+     * Initializes a new instance of the Item class.
+     *
+     * @param gameStarter   FullScreenPokemon instance this is used for.
+     * @param teamAndAction   Team and move being performed.
+     * @param item   Item being used.
+     */
+    public constructor(gameStarter: TGameStartr, teamAndAction: ITeamAndAction<IMoveAction>, item: string) {
+        super(gameStarter);
+
+        this.teamAndAction = teamAndAction;
+        this.item = item;
+
+        const battleInfo: IBattleInfo = gameStarter.battleMover.getBattleInfo() as IBattleInfo;
```

Comment:
Once this PR goes in, please file a followup issue to consolidate this class with the `Move` class. Not your responsibility, but it'll be good code cleanup.

IMO there should be one general class used for both.